internal/core/operations/rooms/add/executor.go

```diff
@@ -88,24 +88,61 @@ func (ex *Executor) Execute(ctx context.Context, op *operation.Operation, defini
 		amount = ex.config.AmountLimit
 	}
 
-	errGroup, errContext := errgroup.WithContext(ctx)
-	executionLogger.Info("start adding rooms", zap.Int32("amount", amount))
+	var wg sync.WaitGroup
+	errsCh := make(chan error, amount)
+
 	for i := int32(1); i <= amount; i++ {
-		errGroup.Go(func() error {
-			err := ex.createRoom(errContext, scheduler, executionLogger)
-			return err
-		})
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			err := ex.createRoom(ctx, scheduler, executionLogger)
+			if err != nil {
+				errsCh <- err
+			}
+		}()
 	}
 
-	if executionErr := errGroup.Wait(); executionErr != nil {
-		executionLogger.Error("Error creating rooms", zap.Error(executionErr))
-		ex.operationManager.AppendOperationEventToExecutionHistory(ctx, op, executionErr.Error())
-		return executionErr
+	go func() {
+		wg.Wait()
+		close(errsCh)
+	}()
```

Comment:
what's the diff between putting this out of a goroutine and block on the `wg.Wait()` rather than blocking in the `range errsCh` below?